sentry-apollo-3/src/test/java/io/sentry/apollo3/SentryApollo3InterceptorWithComposerTest.kt

```diff
@@ -0,0 +1,232 @@
+  "data": {
+    "launch": {
+      "__typename": "Launch",
+      "id": "83",
+      "site": "CCAFS SLC 40",
+      "mission": {
+        "__typename": "Mission",
+        "name": "Amos-17",
+        "missionPatch": "https://images2.imgbox.com/a0/ab/XUoByiuR_o.png"
+      }
+    }
+  }
+}""",
+            socketPolicy: SocketPolicy = SocketPolicy.KEEP_OPEN,
+            beforeSpan: BeforeSpanCallback? = null,
+        ): ApolloClient {
+            whenever(hub.options).thenReturn(SentryOptions())
+
+            server.enqueue(
+                MockResponse()
+                    .setBody(responseBody)
+                    .setSocketPolicy(socketPolicy)
+                    .setResponseCode(httpStatusCode)
+            )
+
+            if (beforeSpan != null) {
+                httpInterceptor = SentryApollo3HttpInterceptor(hub, beforeSpan)
+            }
+
+            val builder = ApolloClient.builder()
+                .networkTransport(
+                    HttpNetworkTransport.Builder()
+                        .httpRequestComposer(SentryApollo3RequestComposer(server.url("/").toString()))
+                        .addInterceptor(httpInterceptor)
+                        .build()
+                )
+
+            return builder.build()
+        }
+    }
+
+    private val fixture = Fixture()
+
+    @Test
+    fun `creates a span around the successful request`() {
+        executeQuery()
+
+        verify(fixture.hub).captureTransaction(
+            check {
+                assertTransactionDetails(it)
+                assertEquals(SpanStatus.OK, it.spans.first().status)
+            },
+            anyOrNull<TraceContext>(),
+            anyOrNull(),
+            anyOrNull()
+        )
+    }
+
+    @Test
+    fun `creates a span around the failed request`() {
+        executeQuery(fixture.getSut(httpStatusCode = 403))
+
+        verify(fixture.hub).captureTransaction(
+            check {
+                assertTransactionDetails(it)
+                assertEquals(SpanStatus.PERMISSION_DENIED, it.spans.first().status)
+            },
+            anyOrNull<TraceContext>(),
+            anyOrNull(),
+            anyOrNull()
+        )
+    }
+
+    @Test
+    fun `creates a span around the request failing with network error`() {
+        executeQuery(fixture.getSut(socketPolicy = SocketPolicy.DISCONNECT_DURING_REQUEST_BODY))
+
+        verify(fixture.hub).captureTransaction(
+            check {
+                assertTransactionDetails(it)
+                assertEquals(SpanStatus.INTERNAL_ERROR, it.spans.first().status)
+            },
+            anyOrNull<TraceContext>(),
+            anyOrNull(),
+            anyOrNull()
+        )
+    }
+
+    @Test
+    fun `when there is no active span, does not add sentry trace header to the request`() {
+        executeQuery(isSpanActive = false)
+
+        val recorderRequest = fixture.server.takeRequest()
+        assertNull(recorderRequest.headers[SentryTraceHeader.SENTRY_TRACE_HEADER])
```

Comment:
Can we add baggage as well please? Same below.

Reply:
If it takes too much time, we can unblock folks first with apollo v3 and add the baggage next.

Reply:
Added in the `SentryApollo3InterceptorTest`